hack/externalnode/install-vm.sh

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eo pipefail
+
+function echoerr {
+    >&2 echo "$@"
+}
+
+_usage="Usage: $0 [--ns <NameSpace>] [--bin <AntreaAgentSavePath>] [--config <AgentConfigSavePath>] [--kubeconfig <KubeconfigSavePath>] [--antrea-kubeconfig <AntreaKubeconfigSavePath>] [--nodename <ExternalNodeName>] [--help|-h]
+        --ns                          NameSpace to be used by the antrea-agent.
+        --bin                         Path of the antrea-agent binary
+        --config                      Path of the antrea-agent configuration file
+        --kubeconfig                  Path of the kubeconfig to access K8s API Server
+        --antrea-kubeconfig           Path of the kubeconfig to access Antrea API Server
+        --nodename                    ExternalNode name to be used by the antrea-agent
+        --help, -h                    Print this message and exit
+
+Please run the script as sudo user"
+
+function print_usage {
+    echoerr "$_usage"
+}
+
+function print_help {
+    echoerr "Try '$0 --help' for more information."
+}
+
+INSTALL_PATH="/usr/sbin"
+AGENT_BIN_PATH=""
+CONFIG_PATH=""
+KUBECONFIG=""
+ANTREAKUBECONFIG=""
+AGENT_NAMESPACE=""
+NODE_NAME="$(hostname)"
+AGENT_LOG_DIR="/var/log/antrea"
+AGENT_CONF_PATH="/etc/antrea"
+# List of supported OS versions, verified by antrea.
+declare -a SUPPORTED_OS=("Ubuntu 18.04", "Ubuntu 20.04")
+
+check_supported_platform() {
+  echo "Checking supported OS platform"
+  dist_version="$(lsb_release -is) $(lsb_release -rs)"
+  for ver in "${SUPPORTED_OS[@]}"; do
+    if [ "$ver" == "$dist_version" ]; then
+      return
+    fi
+  done
+  echoerr "Error ${SUPPORTED_OS[*]} are supported"
```

Comment:
I still believe that for Linux, asking the user whether they want to proceed or not even if their OS is not "verified" would be a better approach. But we can always make that change later.

Reply:
In the interest of time we can take this up later.